### PR TITLE
convert `SocketChannelEndpoint` to `SocketChannel`

### LINKED ISSUE #13771: ERROR async.streaming-response when Jetty SSL enabled
**Describe the bug**
I'm seeing `ERROR async.streaming-response` error messages whenever questions are run. The errors only occur when SSL is enabled as per the [Customizing the Metabase Jetty webserver](https://www.metabase.com/docs/latest/operations-guide/customizing-jetty-webserver.html) instructions. The requests do complete successfully.

I'm aware of the same logs being reported in #12335, but that involved a timeout. I am not seeing any timeouts.

**Logs**
Console log showing the error and the completed ASYNC call:
```bash
2020-11-12 21:37:55,865 DEBUG middleware.log :: GET /api/collection 200 70.2 ms (3 DB calls) App DB connections: 1/4 Jetty threads: 8/50 (3 idle, 0 queued) (57 total active threads) Queries in flight: 0 (0 queued)
2020-11-12 21:37:55,951 DEBUG middleware.log :: GET /api/card/54 200 167.8 ms (6 DB calls) App DB connections: 1/4 Jetty threads: 8/50 (3 idle, 0 queued) (57 total active threads) Queries in flight: 0 (0 queued)
2020-11-12 21:37:56,403 DEBUG middleware.log :: GET /api/alert/question/54 200 24.6 ms (1 DB calls) App DB connections: 1/4 Jetty threads: 8/50 (3 idle, 0 queued) (57 total active threads) Queries in flight: 0 (0 queued)
2020-11-12 21:37:56,457 DEBUG middleware.log :: GET /api/collection 200 69.8 ms (3 DB calls) App DB connections: 1/4 Jetty threads: 8/50 (3 idle, 0 queued) (57 total active threads) Queries in flight: 0 (0 queued)
2020-11-12 21:37:57,517 ERROR async.streaming-response :: Error determining whether HTTP request was canceled
java.lang.ClassCastException: class org.eclipse.jetty.io.SocketChannelEndPoint cannot be cast to class java.nio.channels.SocketChannel (org.eclipse.jetty.io.SocketChannelEndPoint is in unnamed module of loader 'app'; java.nio.channels.SocketChannel is in module java.base of loader 'bootstrap')
	at metabase.async.streaming_response$canceled_QMARK_.invokeStatic(streaming_response.clj:140) [metabase.jar:?]
	at metabase.async.streaming_response$canceled_QMARK_.invoke(streaming_response.clj:131) [metabase.jar:?]
	at metabase.async.streaming_response$start_async_cancel_loop_BANG_$fn__24762$state_machine__15982__auto____24769$fn__24771$inst_24711__24786.invoke(streaming_response.clj:166) [metabase.jar:?]
	at metabase.async.util$cancelable_thread_call$fn__24417$fn__24418.invoke(util.clj:51) [metabase.jar:?]
	at metabase.async.util$cancelable_thread_call$fn__24417.invoke(util.clj:50) [metabase.jar:?]
	at clojure.lang.AFn.applyToHelper(AFn.java:152) [metabase.jar:?]
	at clojure.lang.AFn.applyTo(AFn.java:144) [metabase.jar:?]
	at clojure.core$apply.invokeStatic(core.clj:665) [metabase.jar:?]
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1973) [metabase.jar:?]
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1973) [metabase.jar:?]
	at clojure.lang.RestFn.invoke(RestFn.java:425) [metabase.jar:?]
	at clojure.lang.AFn.applyToHelper(AFn.java:156) [metabase.jar:?]
	at clojure.lang.RestFn.applyTo(RestFn.java:132) [metabase.jar:?]
	at clojure.core$apply.invokeStatic(core.clj:669) [metabase.jar:?]
	at clojure.core$bound_fn_STAR_$fn__5749.doInvoke(core.clj:2003) [metabase.jar:?]
	at clojure.lang.RestFn.invoke(RestFn.java:397) [metabase.jar:?]
	at clojure.lang.AFn.run(AFn.java:22) [metabase.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
2020-11-12 21:37:57,586 DEBUG middleware.log :: POST /api/card/54/query 202 [ASYNC: completed] 1.1 s (6 DB calls) App DB connections: 1/4 Jetty threads: 6/50 (3 idle, 0 queued) (59 total active threads) Queries in flight: 0 (0 queued)
```

Javascript console logs attached:
[metabase-1605219477569.log](https://github.com/metabase/metabase/files/5533597/metabase-1605219477569.log)

**To Reproduce**
1. Run any Metabase question
2. Observe no error messages in the logs
3. Enable SSL with Jetty via: [Customizing the Metabase Jetty webserver](https://www.metabase.com/docs/latest/operations-guide/customizing-jetty-webserver.html)
4. Run any Metabase question
5. See error messages

**Expected behavior**
No error messages.

**Severity**
Low. All working, however errors in the logs are disconcerting.

**Additional context**
Our Metabase instance is run on Kubernetes with PostgreSQL.

**Metabase Diagnostic Info**

```json
{
  "browser-info": {
    "language": "en-CA",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.193 Safari/537.36 Edg/86.0.622.68",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.15.0-1096-azure",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.6"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-10-26",
      "tag": "v0.37.0.2",
      "branch": "release-x.37.x",
      "hash": "ba7be09"
    },
    "settings": {
      "report-timezone": "Asia/Calcutta"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by escherize**
Fixes https://github.com/metabase/metabase/issues/13771

During SSL, the getTransport method was returning a type that we didn't expect: `SocketChannelEndpoint`, and that type was unable to be cast into a `SocketChannel`. Calling `.getChannel` on the `SocketChannelEndpoint` returns a `SocketChannel`, so that is what we do.

### CODE REVIEW

File: src/metabase/async/streaming_response.clj [hunk 1]
```diff
@@ -138,15 +154,15 @@
   immediately, returning `0`."
   [^Request request]
   (try
-    (let [^SocketChannel channel (.. request getHttpChannel getEndPoint getTransport)
-          buf    (ByteBuffer/allocate 1)
-          status (.read channel buf)]
+    (log/fatal "--------------------------------------------------")
```
Comment (dpsutton) [inline src/metabase/async/streaming_response.clj:157]: What is the thinking of this `log/fatal` here?

Comment (escherize) [reply src/metabase/async/streaming_response.clj:157]: . o O (Forgot to remove it)

File: src/metabase/async/streaming_response.clj [hunk 2]
```diff
@@ -130,6 +130,35 @@
...
+  error over and over."
+  (atom #{}))
+
+(defn log-unexpected-transport!
+  "Log an error when an unexpected transport is encountered."
+  [transport]
+  (let [transport-type (type transport)]
+    (when-not (contains? @*reported-types transport-type)
+      (log/errorf "Unexpected transport type encountered in `canceled?`: %s" transport-type))
+    (swap! *reported-types conj transport-type)))
```
Comment (escherize) [inline src/metabase/async/streaming_response.clj:160]: We just want to log this error once per unexpected type

**Review by johnswanson — APPROVED** [review]
Nice, looks good!

**Review by johnswanson — APPROVED** [review]

**Pull request merged**